src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -47,42 +47,39 @@ public boolean addAll(List<T> list) {
 
     @Override
     public T get(int index) {
-        checkPositionForGetAndSet(index);
-        return node(index).elementValue;
+        checkIndexForGetAndSet(index);
+        return getNodeByIndex(index).elementValue;
     }
 
     @Override
     public T remove(int index) {
-        checkPositionForGetAndSet(index);
-        return unlinkNode(node(index));
+        checkIndexForGetAndSet(index);
+        return unlinkNode(getNodeByIndex(index));
     }
 
     @Override
     public boolean remove(T value) {
-        if (value == null) {
-            for (Node<T> nodeForRemove = head;
-                    nodeForRemove != null; nodeForRemove = nodeForRemove.next) {
-                if (nodeForRemove.elementValue == null) {
-                    unlinkNode(nodeForRemove);
-                    return true;
-                }
-            }
-        } else {
-            for (Node<T> nodeForRemove = head;
-                    nodeForRemove != null; nodeForRemove = nodeForRemove.next) {
-                if (value.equals(nodeForRemove.elementValue)) {
-                    unlinkNode(nodeForRemove);
-                    return true;
-                }
+        Node<T> nodeForRemove = head;
+        if (nodeForRemove.next == null) {
+            unlinkNode(nodeForRemove);
+            return true;
+        }
+        while (nodeForRemove.next != null) {
```

Comment:
make while(nodeForRemove == null) then you will not need if()

Reply:
done.